components/list/list-item-drag-mixin.js

```diff
@@ -171,10 +445,12 @@ export class NewPositionEventDetails {
 		// now that we have a reference to the item, shove everything between the
 		// destination to the origin over one
 		if (origin > destination) {
+			destination = this.moveBeforeDestination ? destination : destination + 1;
```

Comment:
Placing before or after the destination was based off of the location compared with the origin. So this allows the component to decide if it goes above or below.